src/Models/seawater_density.jl

```diff
@@ -0,0 +1,132 @@
+SeawaterBuoyancy{Float64}:
+├── gravitational_acceleration: 9.80665
+└── equation of state: BoussinesqEquationOfState{Float64}
+
+julia> model = NonhydrostaticModel(; grid, buoyancy, tracers)
+NonhydrostaticModel{CPU, RectilinearGrid}(time = 0 seconds, iteration = 0)
+├── grid: 32×32×1 RectilinearGrid{Float64, Periodic, Periodic, Flat} on CPU with 3×3×0 halo
+├── timestepper: QuasiAdamsBashforth2TimeStepper
+├── tracers: (S, T)
+├── closure: Nothing
+├── buoyancy: SeawaterBuoyancy with g=9.80665 and BoussinesqEquationOfState{Float64} with ĝ = NegativeZDirection()
+└── coriolis: Nothing
+
+julia> set!(model, S = 34.7, T = 0.5)
+
+julia> density_field = Field(SeawaterDensity(model))
+32×32×1 Field{Center, Center, Center} on RectilinearGrid on CPU
+├── grid: 32×32×1 RectilinearGrid{Float64, Periodic, Periodic, Flat} on CPU with 3×3×0 halo
+├── boundary conditions: FieldBoundaryConditions
+│   └── west: Periodic, east: Periodic, south: Periodic, north: Periodic, bottom: Nothing, top: Nothing, immersed: ZeroFlux
+├── operand: KernelFunctionOperation at (Center, Center, Center)
+├── status: time=0.0
+└── data: 38×38×1 OffsetArray(::Array{Float64, 3}, -2:35, -2:35, 1:1) with eltype Float64 with indices -2:35×-2:35×1:1
+    └── max=0.0, min=0.0, mean=0.0
+
+julia> compute!(density_field)
+32×32×1 Field{Center, Center, Center} on RectilinearGrid on CPU
+├── grid: 32×32×1 RectilinearGrid{Float64, Periodic, Periodic, Flat} on CPU with 3×3×0 halo
+├── boundary conditions: FieldBoundaryConditions
+│   └── west: Periodic, east: Periodic, south: Periodic, north: Periodic, bottom: Nothing, top: Nothing, immersed: ZeroFlux
+├── operand: KernelFunctionOperation at (Center, Center, Center)
+├── status: time=0.0
+└── data: 38×38×1 OffsetArray(::Array{Float64, 3}, -2:35, -2:35, 1:1) with eltype Float64 with indices -2:35×-2:35×1:1
+    └── max=1027.7, min=1027.7, mean=1027.7
+```
+
+Values for `temperature`, `salinity` and `geopotential_height` can be passed to
+`SeawaterDensity` to override the defaults that are obtained from the `model`.
+"""
+function SeawaterDensity(model::ModelsWithBuoyancy;
+                         temperature = model_temperature(model.buoyancy.model, model),
+                         salinity = model_salinity(model.buoyancy.model, model),
+                         geopotential_height = model_geopotential_height(model))
+
+    eos = model.buoyancy.model.equation_of_state
+    validate_model_eos(eos)
+    # Convert function or constant user input to AbstractField
+    grid = model.grid
+    loc = (Center, Center, Center)
+    temperature = field(loc, temperature, grid)
+    salinity = field(loc, salinity, grid)
+    # Preferable here is to leave `geopotential_height` as an `AbstractOperation` rather than creating a `Field`
+    geopotential_height = geopotential_height isa KernelFunctionOperation ? Field(geopotential_height) :
+                                                                            field(loc, geopotential_height, grid)
```

Comment:
```suggestion
    # Preferable here is to leave `geopotential_height` as an `AbstractOperation` rather than creating a `Field`
    geopotential_height = geopotential_height isa KernelFunctionOperation ? geopotential_height :
                                                                            field(loc, geopotential_height, grid)
```

was this intended?

Reply:
Yep that is what I wanted. I thought I tried it and it did not work but seems it does!

Reply:
Possibly we should check `AbstractOperation` rather than just `KernelFunctionOperation` (a minor niceity)

Reply:
Done, tests also now check `seawater_density(model) isa AbstractOperation` as well. Is that preferable?